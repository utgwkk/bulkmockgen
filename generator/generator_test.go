package generator

import (
	"strings"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func TestGenerate(t *testing.T) {
	testcases := []struct {
		name string
		g    *Generator
	}{
		{
			name: "multi file",
			g: &Generator{
				UseGoRun:      false,
				PackageName:   "mock_multifile",
				SourcePackage: "fixtures/multifile/",
				MockSetName:   "MockInterfaces",
			},
		},
		{
			name: "multi mock set",
			g: &Generator{
				UseGoRun:      false,
				PackageName:   "mock_multiset",
				SourcePackage: "fixtures/multiset/",
				MockSetName:   "MockFoos",
			},
		},
	}
	for _, tc := range testcases {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			var sb strings.Builder
			tc.g.writer = &sb

			err := tc.g.Generate()
			if err != nil {
				t.Fatal(err)
			}

			snaps.MatchSnapshot(t, sb.String())
		})
	}
}
