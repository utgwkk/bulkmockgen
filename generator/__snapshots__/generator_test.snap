
[TestGenerate/multi_file - 1]
package multifile
// Code generated by mockgengen. DO NOT EDIT.

//go:generate mockgen -destination mock_multifile/mock_multifile.go -package mock_multifile . IFoo,IBar

---

[TestGenerate/multi_mock_set - 1]
package multiset
// Code generated by mockgengen. DO NOT EDIT.

//go:generate mockgen -destination mock_multiset/mock_multiset.go -package mock_multiset . IFoo

---
